templates/oauth2/authorize.html

```diff
@@ -213,13 +220,22 @@
 
     <p>Please confirm that the application <strong>${client.description}</strong>
       (by <a href="/~${LOGIN(client.owner.profile.username)}">${client.owner.profile.username}</a>)
-      should be authorized to make requests on your behalf.</p>
+      should be authorized to make requests on your behalf with the following permissions:</p>
+
+    $if scopes:
+      <table id="permissions">
+        <tr><td><strong>PERMISSION</strong></td><td><strong>GRANTS</strong></td></tr>
+        $for scope in scopes:
+          <tr><td>${scope['name']}</td><td>${scope['description']}</td></tr>
+      </table>
+    $else:
+      <p>No elevated permissions requested</p>
 
-    <p>Note that this means that <strong>${client.description}</strong> can
-      perform <em>almost any action</em> as if you were logged in. Do
-      not authorize this application if you do not trust it. However, this
+    <p>
+      Do not authorize this application if you do not trust it. However, this
       application will <em>never</em> be able to see or change your
```

Comment:
Why is this second sentence necessary? I'd rather people err on the side of paranoia.